packages/react-router-dom/modules/Focus.js

```diff
@@ -0,0 +1,59 @@
+import React from "react";
+import { __RouterContext as RouterContext } from "react-router";
+import warning from "tiny-warning";
+
+class FocusWithLocation extends React.Component {
+  setRef = element => {
+    this.eleToFocus = element;
+  };
+
+  render() {
+    const { children } = this.props;
+    return children(this.setRef);
+  }
+
+  componentDidMount() {
+    this.focus();
+  }
+
+  componentDidUpdate(prevProps) {
+    // only re-focus when the location changes
+    if (this.props.location !== prevProps.location) {
+      this.focus();
+    }
+  }
+
+  focus() {
+    const { preserve, preventScroll = false } = this.props;
```

Comment:
Let's put the `preventScroll` default value in `Focus.defaultProps`, [like this](https://github.com/ReactTraining/react-router/blob/5ccb241cd59eae7a15de31c7babff6619378be39/packages/react-router/modules/Redirect.js#L61-L63).